### PR TITLE
Feature/#8

### PR DESCRIPTION
**Pull request by SukritGupta17**
Features to deregister, stop task for AWS and terminate pod/deployment for GKE

### CODE REVIEW

File: src/ucapi/gke_client.py [hunk 1]
```diff
@@ -700,11 +711,61 @@ def run_task(self, task_definition, cluster_name):
 
         return response
 
-    def deregister_task(self):
-        pass
+    def stop_task(self,task_definition, cluster_name=None):
+        """
+        Takes the task definition as dictionary (JSON formatted)
+        or as a filepath to a YAML file. Additionally, can take a pod definition as a YAML/equivalent format and
```
Comment (callanmch) [inline src/ucapi/gke_client.py]: Why does this sentence just end?

File: src/ucapi/gke_client.py [hunk 2]
```diff
@@ -700,11 +711,61 @@ def run_task(self, task_definition, cluster_name):
...
+        elif task_definition_yaml['apiVersion'] == 'v1':
+            k8s = client.CoreV1Api()
+        else:
+            raise "API Client {} has not been implemented yet.".format(task_definition_yaml['apiVersion'])
+        # Selects the kind from YAML
+        if task_definition_yaml['kind'] == 'Pod':
+            try:
+                # Refer to the Pod by name, Need to add its namespace and pass some delete options.
+                pod_name = task_definition_yaml.get('metadata',{}).get('name')
+                response = k8s.delete_namespaced_pod(name=pod_name, namespace="default", body=client.V1DeleteOptions())
```
Comment (callanmch) [inline src/ucapi/gke_client.py]: what if the namespace should be something other than default?

File: src/ucapi/gke_client.py [hunk 3]
```diff
@@ -700,11 +711,61 @@ def run_task(self, task_definition, cluster_name):
...
+                print('Pod terminated. \nstatus={}\n'.format(response.status))
+            except:
+                raise InvalidTerminationException("Exception when deleting pod on {}"
+                                                 .format(task_definition_yaml['apiVersion']))
+        elif task_definition_yaml['kind'] == 'Deployment':
+            try:
+                # only works for extensionV1beta1api
+                deployment_name = task_definition_yaml.get('metadata',{}).get('name')
+                response = k8s.delete_namespaced_deployment(name=deployment_name, namespace="default", body=client.V1DeleteOptions())
```
Comment (callanmch) [inline src/ucapi/gke_client.py]: what if the namespace should be something other than default?

File: src/ucapi/gke_client.py [hunk 4]
```diff
@@ -700,11 +711,61 @@ def run_task(self, task_definition, cluster_name):
...
+            k8s = client.CoreV1Api()
+        else:
+            raise "API Client {} has not been implemented yet.".format(task_definition_yaml['apiVersion'])
+        # Selects the kind from YAML
+        if task_definition_yaml['kind'] == 'Pod':
+            try:
+                # Refer to the Pod by name, Need to add its namespace and pass some delete options.
+                pod_name = task_definition_yaml.get('metadata',{}).get('name')
+                response = k8s.delete_namespaced_pod(name=pod_name, namespace="default", body=client.V1DeleteOptions())
+                print('Pod terminated. \nstatus={}\n'.format(response.status))
```
Comment (callanmch) [inline src/ucapi/gke_client.py]: We don't need to be doing any printing on behalf of the users.

File: src/ucapi/gke_client.py [hunk 5]
```diff
@@ -700,11 +711,61 @@ def run_task(self, task_definition, cluster_name):
...
+            except:
+                raise InvalidTerminationException("Exception when deleting pod on {}"
+                                                 .format(task_definition_yaml['apiVersion']))
+        elif task_definition_yaml['kind'] == 'Deployment':
+            try:
+                # only works for extensionV1beta1api
+                deployment_name = task_definition_yaml.get('metadata',{}).get('name')
+                response = k8s.delete_namespaced_deployment(name=deployment_name, namespace="default", body=client.V1DeleteOptions())
+                print('Deployment terminated. \nstatus{}\n'.format(response.status))
```
Comment (callanmch) [inline src/ucapi/gke_client.py]: We don't need to be doing any printing on behalf of the users.

Comment (SukritGupta17) [reply src/ucapi/gke_client.py]: fixed

File: src/ucapi/gke_client.py (hunk 4 above)
Comment (SukritGupta17) [reply src/ucapi/gke_client.py]: fixed

File: src/ucapi/gke_client.py (hunk 1 above)
Comment (SukritGupta17) [reply src/ucapi/gke_client.py]: fixed

File: src/ucapi/gke_client.py (hunk 2 above)
Comment (SukritGupta17) [reply src/ucapi/gke_client.py]: the use of multiple namespaces is an advanced usage, it is used when multiple groups/teams are working on the same cluster but want varying access policies for each group and have isolation. To create multiple namespaces the user needs to create a namespace YAML file Explicitly. No idea how to handle all that.

**Pull request merged**